getdeps.sh

```diff
@@ -62,14 +62,31 @@ function build_cmake() {
     )
 }
 
+function build_gtest_and_gmock() {
+    (
+        for LIB in gmock gtest
+        do
+          echo "building $LIB..."
+          cd "/usr/src/$LIB" || die "Missing dir /usr/src/$LIB"
+          if [ -e ./CMakeLists.txt ]; then
+              echo cmake .. $CMAKEFLAGS
+              sudo cmake CMakeLists.txt $CMAKEFLAGS || die "Cmake failed for $LIB"
+              sudo make -j "$NPROC" || die "Failed to build $LIB"
```

Comment:
This looks like you're creating the libraries and .o files in /usr/src, which is probably not the right thing.  What about something like this instead:

```
mkdir build.test && cd build.test
for LIB in gmock gtest ; do
          mkdir $LIB && cd $LIB
           echo "building $LIB..."
           test -d /usr/src/$LIB || die "Missing dir /usr/src/$LIB"
           cmfile=/usr/src/$LIB/CMakeLists.txt
           test -e $cmfile || dir "Missing $cmfile"
           cmake $cmfile $CMAKEFLAGS || die "Cmake failed for $LIB"
           make -j "$NPROC" || die "Failed to build $LIB"
       end
```